python/tank/bootstrap/cached_configuration.py

```diff
@@ -239,6 +241,10 @@ def update_configuration(self):
                     os.path.join(self._path.current_os, "install", "core")
                 )
                 log.debug("Previous core restore complete...")
+        else:
+            # remove backup folders now that the update has completed successfully
+            filesystem.delete_folder(os.path.dirname(config_backup_path))
```

Comment:
i would probably put a try/catch around these too just to make sure that a user never gets blocked because the cleanup failed. Catch it, log it as an error and continue.